src/ape/types/__init__.py

```diff
@@ -226,6 +193,57 @@ def __getitem__(self, item: str) -> Any:
     def get(self, item: str, default: Optional[Any] = None) -> Any:
         return self.event_arguments.get(item, default)
 
+    class Config:
+        # NOTE: Due to https://github.com/samuelcolvin/pydantic/issues/1241 we have
+        # to add this cached property workaround in order to avoid this error:
+
+        #    TypeError: cannot pickle '_thread.RLock' object
+
+        keep_untouched = (cached_property,)
+
+
+class ContractLog(EventArguments):
```

Comment:
I had originally planned a completely different object in my head and started there. This could really just be moved to the ContractLog, and keep ContractLog a BaseModel. Let me know if you want to do that after looking at the latest changes